sapp/ui/frontend/src/Runs.js

```diff
@@ -38,7 +39,29 @@ function Run(props: $ReadOnly<{run: RunDescription}>): React$Node {
     );
   };
 
-  return (
+  const deleteRunMutation = gql`
+    mutation DeleteRun($id: ID!) {
+      delete_run(input: {id: $id}) {
+        clientMutationId
+      }
+    }
+  `;
+
+  const [deleteRun, {error: deleteError}] = useMutation(
+    deleteRunMutation
+  );
+
+  const onDelete = (): void => {
+    deleteRun({variables: {id: props.run.run_id}});
+    setShowRun(false);
+  };
+
+  if(deleteError) {
+    Modal.error({title: 'Unable to delete run ', content: deleteError.toString()});
+    return null;
+  }
```

Comment:
Disclaimer: react noob here!
Shouldn't this be inside the `onDelete` function? How could `deleteError` have a value before it is actually called by `onDelete`?

Reply:
Hey, sorry for the late reply. I am not that familiar in React myself as well, so even though I knew how it works, I wanted to be sure. deleteError is derived from useMutation() and is a react hook [https://www.apollographql.com/docs/tutorial/mutations/#apply-the-usemutation-hook] 

So hooks alter the flow of the program when their value changes in a nutshell, so when you change its value all control instructions that use it are re-evaluated. Hence, we get the error as per the screenshot :) I reproduced this by modifying the graphene query to some illogical mutation name...

<img width="1440" alt="Screenshot 2021-07-16 at 4 17 50 PM" src="https://user-images.githubusercontent.com/8947010/125938255-6d6bb2bf-1412-4816-96dd-12d2ade85066.png">

[Ref to hooks: https://reactjs.org/docs/hooks-intro.html]

Reply:
But yeah, thanks for this comment, I did a hard testing of the error message and found there was a slight bug that the run disappeared (to appear on reload) even incase of error. Fixed that in the latest change. So, thank you!